progs/get-subseqs/main.go

```diff
@@ -30,5 +56,43 @@ func main() {
 		defer coordFile.Close()
 	}
 
-	fmt.Println("I should be processing the coordinate stream now!")
+	//######################
+
+	output := ""
+	fileScannerCoord := bufio.NewScanner(coordFile)
+
+	for fileScannerCoord.Scan() {
+		lineCoord := fileScannerCoord.Text()
+		if len(lineCoord) == 0 {
+			continue
+		}
+		split_str := strings.Fields(lineCoord)
+		if len(split_str) != 3 {
+			fmt.Println("Error, length of coordinates")
+		}
+		for i, v := range split_str {
+			split_str[i] = strings.TrimSpace(v)
+		}
+
+		for _, v := range sequences {
+
+			if strings.HasPrefix(v, split_str[0]) {
+
+				basepair := strings.TrimSpace(strings.TrimPrefix(v, split_str[0]))
+				low, er1 := strconv.Atoi(split_str[1])
+				high, er2 := strconv.Atoi(split_str[2])
+				if er1 != nil || er2 != nil {
+					fmt.Printf("Error")
+
+				}
+				output = output + basepair[low-1:high-1] + "\n"
```

Comment:
Quadratic in the output here. Concatenating strings is a female dog when it comes to performance...